src/frequenz/sdk/model/_handler.py

```diff
@@ -0,0 +1,448 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Load, update, monitor and retrieve machine learning models."""
+
+from __future__ import annotations
+
+import abc
+import asyncio
+import bisect
+import calendar
+import logging
+import os
+import pickle
+from dataclasses import dataclass
+from datetime import datetime
+from pathlib import Path
+from typing import Generic, Type, TypeVar
+
+from frequenz.channels.util import FileWatcher, Select
+
+from frequenz.sdk._internal._asyncio import cancel_and_await
+
+_logger = logging.getLogger(__name__)
+
+T = TypeVar("T")
+
+
+@dataclass
+class ModelKeyPolicy(abc.ABC):
+    """A policy to handle mappings from model periods to number keys."""
+
+    _period_to_num: dict[str, int]
+    """Mappings of model periods to number keys
```

Comment:
This doesn't apply to the `SingleModelKeyPolicy`, so I wouldn't put it in the abstract base class.